ui/src/app/domains/views/DomainsList/DomainListHeader/DomainListHeader.tsx

```diff
@@ -13,20 +15,42 @@ const DomainListHeader = (): JSX.Element => {
   const domainCount = useSelector(domainSelectors.count);
   const domainsLoaded = useSelector(domainSelectors.loaded);
 
+  const [isFormOpen, setFormOpen] = useState(false);
+
   useEffect(() => {
     dispatch(domainActions.fetch());
   }, [dispatch]);
 
+  let buttons = [
+    <Button
+      appearance="neutral"
+      data-test="add-domain"
+      key="add-domain"
+      onClick={() => setFormOpen(true)}
+    >
+      Add domains
+    </Button>,
+  ];
+
+  let formWrapper = null;
+
+  if (isFormOpen) {
+    buttons = null;
```

Comment:
I think this is causing a strict mode error because the type is inferred to be `JSX.Element[]` above when `buttons` is defined. You'll need to strictly define the type as `let buttons: JSX.Element[] | null = [`.

I'm surprised there isn't a similar error for `formWrapper` though. Maybe the ts compiler is smart enough to know that something defined as `null` is prooooobably not going to stay as `null`.